docs/design/persistent-storage.md

```diff
@@ -0,0 +1,235 @@
+# Persistent Stores
+
+This document proposes a model for managing persistent, cluster-scoped storage for applications requiring long lived data.
+
+### tl;dr
+
+`PersistentStore` objects contain a VolumeSource with a lifecycle greater than the pods it is used by.  Many types of VolumeSources can be implemented as volume plugins, with each implementing an interface for use within a controller.  A pod with persistence requirements (specifically pod.spec.volumes.volume.persistenceRequirements) is bound to a matching VolumeSource from a pool of available stores.  PersistentStore objects give that VolumeSource longevity.
```

Comment:
I have read this like 10 times and I am just not clear.

Is a PersistentStore a single PD, a pool of identical PDs, a pool of heterogeneous PDs, or a descriptor of hoe to make PDs on the fly?

It feels like you're desperately trying to avoid saying "volume" but that is maybe the best word...

The existing meaning of VolumeSource is a description, not a thing, so reading "a VolumeSource with a lifecycle" and "a matching VolumeSource" just doesnt' parse for me.